gammapy/spectrum/reflected.py

```diff
@@ -414,3 +419,124 @@ def plot(self, fig=None, ax=None, cmap=None, idx=None, add_legend=False):
             ax.legend(handles=handles)
 
         return fig, ax, cbar
+
+
+class ReflectedRegionsBackgroundMaker:
+    """Reflected regions background maker.
+
+    Parameters
+    ----------
+    region: `~regions.SkyRegion`
+        On region to compute off regions for.
+    angle_increment : `~astropy.coordinates.Angle`, optional
+        Rotation angle applied when a region falls in an excluded region.
+    min_distance : `~astropy.coordinates.Angle`, optional
+        Minimal distance between two consecutive reflected regions
+    min_distance_input : `~astropy.coordinates.Angle`, optional
+        Minimal distance from input region
+    max_region_number : int, optional
+        Maximum number of regions to use
+    exclusion_mask : `~gammapy.maps.WcsNDMap`, optional
+        Exclusion mask
+    binsz : `~astropy.coordinates.Angle`
+        Bin size of the reference map used for region finding. Default : 0.01 deg
+    """
+
+    def __init__(
+        self,
+        region,
+        angle_increment="0.1 rad",
+        min_distance="0 rad",
+        min_distance_input="0.1 rad",
+        max_region_number=10000,
+        exclusion_mask=None,
+        binsz="0.01 deg",
+    ):
+        self.region = region
+        self.binsz = binsz
+        self.exclusion_mask = exclusion_mask
+        self.angle_increment = Angle(angle_increment)
+        self.min_distance = Angle(min_distance)
+        self.min_distance_input = Angle(min_distance_input)
+        self.max_region_number = max_region_number
+
+    def _get_finder(self, observation):
+        finder = ReflectedRegionsFinder(
```

Comment:
Another coding style suggestion. If a temp variable is just created and returned, but never used, I suggest to put `return` directly and avoid the temp variable. I've changed to that coding pattern already mostly throughout Gammapy

Reply:
Done